src/bin/sccache-dist/main.rs

```diff
@@ -67,18 +73,31 @@ fn main() {
             }
         },
     };
-
-    std::process::exit(match run(command) {
-        Ok(s) => s,
-        Err(e) => {
-            eprintln!("sccache-dist: error: {}", e);
-
-            for e in e.chain().skip(1) {
-                eprintln!("sccache-dist: caused by: {}", e);
+    
+    let run_fut = async move {
+        match run(cmd).await {
+            Ok(s) => s,
+            Err(e) => {
+                eprintln!("sccache-dist: error: {}", e);
+                for e in e.chain().skip(1) {
+                    eprintln!("sccache-dist: caused by: {}", e);
+                }
+                2
             }
-            2
         }
-    });
+    };
+        
+    // Whenever `docker` or `systemd` stop the service, the signals
+    // have to be processed and the app has to perform a graceful stop.
+    select! {
```

Comment:
By the default select will randomly pick the order of polling. You can make it use the order you have specified (also avoiding calling random) by adding `biased`.
I'm on my mobile but it should look something like this:
```suggestion
    select! {
        biased;
        
```